HeroVault/HeroVault/Features/Home/HomeScreen/HomeScreenViewModel.swift

```diff
@@ -0,0 +1,136 @@
+    }
+    
+    // MARK: Functions
+    
+    func set(marvelDataType: EntityType) {
+        self.marvelDataType = marvelDataType
+    }
+    
+    func fetchMarvelNameAndImage(for marvelIndex: Int) -> (String, String) {
+        isSearching ?
+        (filteredMarvelData[marvelIndex].name, "\(filteredMarvelData[marvelIndex].thumbnail)/portrait_incredible.jpg".convertToHttps()) :
+        (marvelData[marvelIndex].name, "\(marvelData[marvelIndex].thumbnail)/portrait_incredible.jpg".convertToHttps())
+    }
+    
+    func fetchMarvelData(atIndex: Int) -> MarvelData? {
+        isSearching ? filteredMarvelData[atIndex] : marvelData[atIndex]
+    }
+    
+    func fetchMarvelData() {
+        marvelDataType == .character ? fetchCharacters() : fetchComics()
+    }
+    
+    func filterMarvelData(filteredText: String) {
+        
+        isSearching = true
+        guard !filteredText.isEmpty else {
+            isSearching = false
+            hideNoResultsText = true
+            return
+        }
+        
+        filteredMarvelData = marvelData.filter { marvelItem in
+            marvelItem.name.lowercased().contains(filteredText.lowercased())
+        }
+        
+        loadSearchNotFoundText()
+        delegate?.reloadView()
+    }
+    
+    func handleSegmentedControl(segmentedControlTitle: String) {
+        isSearching = false
+        segmentedControlTitle == "Characters" ? set(marvelDataType: EntityType.character) : set(marvelDataType: EntityType.comic)
+        fetchMarvelData()
+    }
+    
+    private func fetchCharacters() {
+        marvelData = []
+        homeScreenRepository?.fetchCharacters { [weak self] result in
+            guard let self else { return }
+            switch result {
+            case .success(let characters):
+                for character in characters.data.results {
+                    if !character.name.isEmpty && !character.overview.isEmpty && !checkIfImageIsAvailable(thumbnail: character.thumbnail) {
+                        marvelData.append(character)
+                    }
+                }
+                delegate?.reloadView()
+            case .failure(let error):
+                self.error = error
+            }
+        }
+    }
+    
+    private func fetchComics() {
+        marvelData = []
+        homeScreenRepository?.fetchComics { [weak self] result in
+            guard let self else { return }
+            switch result {
+            case .success(let comics):
+                for comic in comics.data.results {
+                    if !comic.name.isEmpty && !checkIfImageIsAvailable(thumbnail: comic.thumbnail) {
+                        marvelData.append(comic)
+                    }
+                }
+                delegate?.reloadView()
+            case .failure(let error):
+                self.error = error
+            }
+        }
+    }
+    
+    private func checkIfImageIsAvailable(thumbnail: String) -> Bool {
```

Comment:
```suggestion
    private func isImageAvailable(thumbnail: String) -> Bool {
```
When naming functions/ properties you shouldn't include unnecessary words